src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -135,25 +164,213 @@ void ReportHandler(DeliveryReport<string, byte[]> report)
+        QueueBrokerConnectionHelper.SetupKafkaClientConfig(config, Configuration.Connection.KafkaConnectionSettings);
+        return new AdminClientBuilder(config).Build();
+    }
+
+
+    private static IEnumerable<AclBinding> GetMatchingAllowAclBindings(IAdminClient adminClient, AclOperation operation = AclOperation.All,
+        ResourceType resourceType = ResourceType.Any)
+    {
+        var aclBindingsForTransactionIdsTask = adminClient.DescribeAclsAsync(new AclBindingFilter
+        {
+            EntryFilter = new AccessControlEntryFilter {Operation = operation, PermissionType = AclPermissionType.Allow},
+            PatternFilter = new ResourcePatternFilter {ResourcePatternType = ResourcePatternType.Any, Type = resourceType}
+        });
+
+        aclBindingsForTransactionIdsTask.Wait();
+        return aclBindingsForTransactionIdsTask.Result.AclBindings;
+    }
+
+    private bool TryToGetAclErrorsDetails(out string message, string topicName = null)
+    {
+        message = "";
+        try
+        {
+            using var adminClient = GetAdminClient();
+
+            // unknown resource type is assigned to returned transactionalID ACL bindings, but we cannot directly ask API for unknown type resources, so we have to filter.. 
+            var transactionIdsWriteAclBindings = GetMatchingAllowAclBindings(adminClient, AclOperation.Write).Where(x => x.Pattern.Type == ResourceType.Unknown);
+            var transactionIdsDescribeAclBindings = GetMatchingAllowAclBindings(adminClient, AclOperation.Describe).Where(x => x.Pattern.Type == ResourceType.Unknown);
+            
+            bool validTransactionWriteAcl =
+                transactionIdsWriteAclBindings.Any(transactionIdsAclBinding => AssertAclMatchingName(TransactionalId, transactionIdsAclBinding));
+            bool validTransactionDescribeAcl =
+                transactionIdsDescribeAclBindings.Any(transactionIdsAclBinding => AssertAclMatchingName(TransactionalId, transactionIdsAclBinding));
+            
+            
+            bool validTopicWriteAcl = true;
+            if (topicName is not null)
             {
-                _producer.AbortTransaction();
+                var topicsWriteAclBindings = GetMatchingAllowAclBindings(adminClient, AclOperation.Write, ResourceType.Topic);
+                validTopicWriteAcl = topicsWriteAclBindings.Any(topicsAclBinding => AssertAclMatchingName(topicName, topicsAclBinding));    
             }
-            catch (Exception e)
+            
+            if (validTransactionWriteAcl && validTopicWriteAcl && validTransactionDescribeAcl)
             {
-                if (Logger.IsOperationsEnabled)
-                    Logger.Operations($" ETL process: {Name}. Aborting Kafka transaction failed.", e);
+                return false;
             }
 
-            throw new QueueLoadException(ex.Message, ex);
+            var msgBuilder = new StringBuilder($"ETL process: {Name}. Failed to load data for the producer instance. " +
```

Comment:
"ETL process: {Name}. Failed to load data for the producer instance." ->should it be here? This method should return only the details of missing ACL bindings, shouldn't it?

Reply:
removed